### PR TITLE
Adding postgres v16 to application tests

### PR DESCRIPTION
**Pull request by MatthewWeppenaar**

### CODE REVIEW

**Review by KaitCrawford — COMMENTED** [review]
I don't believe this is sufficient @MatthewWeppenaar 
Right now you've just added a line to install postgresql-16, it's not actually being used for the tests.
This needs to be added to a matrix that will manage the postgres versions and run all the steps for each version

**Comment by MatthewWeppenaar** [comment]
What would the env variables for this be, or where can i find it?  Would it be `POSTGRES_DB: reminder_scheduler` ,`POSTGRES_USER: postgres`,` POSTGRES_PASSWORD: postgres`

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -6,6 +6,20 @@ on:
...
+      postgres:
+        image: postgres:${{ matrix.postgres_version }}
+        env:
+          POSTGRES_DB: reminder_scheduler
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+        ports:
+          - 5432:5432
+    env:
+      RP_SIDEKICK_DATABASE: postgres://postgres:postgres@localhost/reminder_scheduler
```
Comment (KaitCrawford) [inline .github/workflows/test.yaml:22]: this must be `DATABASE_URL` please

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -18,6 +32,7 @@ jobs:
           wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc 2>/dev/null | sudo apt-key add -
           sudo apt-get update
           sudo apt-get -y install postgresql-14
+          sudo apt-get -y install postgresql-16
```
Comment (KaitCrawford) [inline .github/workflows/test.yaml:35]: You can remove this line and the one above it. The `services` section of the yaml now makes sure this is being installed

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -18,6 +32,7 @@ jobs:
...
           pip install -r requirements.txt -r requirements-dev.txt
       - name: Run tests
         run: pytest
```
Comment (KaitCrawford) [inline .github/workflows/test.yaml:38]: Please change this to 
```
        run: |
          python manage.py makemigrations --check
          pytest
```

**Review by KaitCrawford — COMMENTED** [review]
Some new comments to ensure that the database is being properly tested

File: .github/workflows/test.yaml (hunk 2 above)
Comment (KaitCrawford) [reply .github/workflows/test.yaml:35]: you can actually replace this whole `run:` with 
`run: pip install -r requirements.txt -r requirements-dev.txt`

**Review by KaitCrawford — APPROVED** [review]
👍 🍰

**Pull request merged**